modules/shared/src/test/scala/org/tessellation/schema/generators.scala

```diff
@@ -118,7 +121,11 @@ object generators {
       txnSalt <- transactionSaltGen
     } yield Transaction(src, dst, txnAmount, txnFee, txnReference, txnSalt)
 
-  val signatureGen: Gen[Signature] = nesGen(str => Signature(Hex(str)))
+  val signatureGen: Gen[Signature] =
+    Gen
+      .listOfN(142, Gen.alphaChar)
```

Comment:
Are we sure that the signature is always 142 chars long? I remember that sometimes signatures seemed to have different length.

Now that I quickly examine the snapshot signatures some of them are longer e.g. 144

Reply:
Interesting. I checked many signatures and it was always `142`. If the signature size is not fixed then the whole signature size prediction makes no sense. Eventually we can take `max(allSignatureSizes)` 🤷🏻‍♂️